atc/builds/tracker_runner_test.go

```diff
@@ -62,85 +70,121 @@ var _ = Describe("TrackerRunner", func() {
 
 	AfterEach(func() {
 		process.Signal(os.Interrupt)
-		<-process.Wait()
+		Eventually(process.Wait()).Should(Receive())
 	})
 
-	It("tracks immediately", func() {
-		<-tracked
+	Context("when the interval elapses", func() {
+
+		JustBeforeEach(func() {
+			fakeClock.WaitForWatcherAndIncrement(interval)
+		})
+
+		Context("when the component is paused", func() {
```

Comment:
we have this context (when the component is paused) in multiple contexts. Can we pull it out into a helper?

Reply:
The reason these kind of logic all over each component (and thus the test) is the pipeline syncer (who spins off a scheduler for each pipeline). Since after we have algorithm-v3 merging in, pipeline syncer wil be removed and a runner for job level scheduler would be introduced, we would then be able to extract the component paused and interval elapsed logic to a centralize place. In that case the duplication will be gone.

So for now we would rather just let it be and hopelly clean it up as soon as algorithm-v3 is there.